flatland/agents/agent.py

```diff
@@ -87,13 +161,26 @@ def add_sensor(self, sensor_type, sensor_name, sensor_config = None, **sensor_pa
         sensor_params['name'] = sensor_name
         sensor_params['type'] = sensor_type
 
-        new_sensor = sensor.SensorGenerator.create(sensor_type, self.frame.anatomy, sensor_params)
+
+        #Brait
+        if sensor_type is 'lidar':
+            new_sensor = geometric_sensor.LidarSensor(self.frame.anatomy, sensor_params)
+            self.has_geometric_sensor = True
+        else:
+            new_sensor = sensor.SensorGenerator.create(sensor_type, self.frame.anatomy, sensor_params)
```

Comment:
In the `sensor` package, a generator is used as a way to create decorators. In short, that means when you give the string parameter "depth" to the generator, it returns you the DepthSensor class. The string and the class are two radically different object. I had no use for now to create a new generator because we have only one type of geometric sensor, and I can't add it to the SensorGenerator because it is internal to the `sensor` package. This will play a role when we create more than one geometric sensor, and it won't be hard at all to make. Should I do it now?

Reply:
So the registration through decorator was a design choice that allows people to install the library, but still be able to define their own entities / sensors / plyaground and register them dynamically without having to reinstall everthing all the time.